src/test/scala/nl/knaw/dans/easy/pid/PidGeneratorAppSpec.scala

```diff
@@ -148,8 +147,22 @@ class PidGeneratorAppSpec extends TestSupportFixture
     start.countDown()
     done.await()
 
+    val expectedResults = Set(
+      "10.5072/dans-2ap-4qfd",
+      "10.5072/dans-zve-22y5",
+      "10.5072/dans-x5f-3p9r",
+      "10.5072/dans-xnr-c7jf",
+      "10.5072/dans-x75-qa68",
+      "10.5072/dans-229-ftfq",
+      "10.5072/dans-zyq-5znp",
+      "10.5072/dans-z8c-gnm3",
+      "10.5072/dans-2zt-warx",
+      "10.5072/dans-z7p-ebra",
+    )
+
     results.keys should contain theSameElementsAs (1 to n).map(i => s"test$i")
-    all(results.values).shouldBe(a[Success[_]])
+    all(results.values) shouldBe a[Success[_]]
+    results.collect { case (_, Success(doi)) => doi }.toSeq should contain theSameElementsAs expectedResults
```

Comment:
I don't really see why concurrent initialize and generate calls result in any problems at all:

* When `generate` gets to the barrier first, it runs [`database.getSeed`](https://github.com/DANS-KNAW/easy-pid-generator/blob/c2a68e21cd8e0fa739884245486e02dc501bd39d/src/main/scala/nl/knaw/dans/easy/pid/generator/PidManagerComponent.scala#L43), which does a [`SELECT ... FOR UPDATE`](https://github.com/DANS-KNAW/easy-pid-generator/blob/c2a68e21cd8e0fa739884245486e02dc501bd39d/src/main/scala/nl/knaw/dans/easy/pid/generator/DatabaseComponent.scala#L36), thereby locking the table for later updates. It will then find the seed actually not being present at all and hence it will [fail immediately without proceeding any further](https://github.com/DANS-KNAW/easy-pid-generator/blob/c2a68e21cd8e0fa739884245486e02dc501bd39d/src/main/scala/nl/knaw/dans/easy/pid/generator/PidManagerComponent.scala#L56). This will release the lock (transaction finished) and make way for `initialize` to do its thing. This command will also run [`database.getSeed`](https://github.com/DANS-KNAW/easy-pid-generator/blob/c2a68e21cd8e0fa739884245486e02dc501bd39d/src/main/scala/nl/knaw/dans/easy/pid/generator/PidManagerComponent.scala#L79) and hence acquiring the same lock as `generate` did. If the seed is indeed not present, it will [initialize the seed](https://github.com/DANS-KNAW/easy-pid-generator/blob/c2a68e21cd8e0fa739884245486e02dc501bd39d/src/main/scala/nl/knaw/dans/easy/pid/generator/PidManagerComponent.scala#L82).
* Given that both `initialize` and `generate` use the same lock, it also works fine when `initialize` gets to the barrier first. In this instance, it initializes the PID, while `database.getSeed` has to wait for the the lock to be released.